perl-lib/OESS/lib/OESS/MPLS/Device/Juniper/MX.pm

```diff
@@ -1217,26 +1217,34 @@ sub get_config_to_remove{
 
             if($type eq 'L2VPN'){
                 my $sites = $xp->find(' ./c:protocols/c:l2vpn/c:site', $ri);
-		foreach my $site (@$sites){
-		    my $site_name = $xp->findvalue( './c:name', $site);
-		    $self->{'logger'}->debug("Site Name: " . $site_name);
-		    my $ints = $xp->find('./c:interface', $site);
-
-		    foreach my $int (@$ints){
-			my $int_full_name = $xp->findvalue( './c:name', $int);
-			my ($int_name,$unit_name) = split('.',$name);
-			$self->{'logger'}->debug("Checking to see if port: $name is part of circuit: $circuit_id");
-			if(!$self->_is_circuit_on_port($circuit_id, $circuits, $int_name, $unit_name)){
+                foreach my $site (@$sites){
+                    my $site_name = $xp->findvalue( './c:name', $site);
+                    my $site_id = $xp->findvalue( './c:site-identifier', $site);
+
+                    my $expected_site_name = $self->{'name'} . '-' . $circuit_id;
+                    my $expected_site_id = $circuits->{$circuit_id}->{site_id};
+
+                    if ($expected_site_id != $site_id || $expected_site_name ne $site_name) {
+                        $ri_dels .= "<instance><name>$name</name><protocols><l2vpn><site operation='delete'><name>$site_name</name></site></l2vpn></protocols></instance>";
+                        next;
+                    }
+
+                    my $ints = $xp->find('./c:interface', $site);
+                    foreach my $int (@$ints){
+                        my $int_full_name = $xp->findvalue( './c:name', $int);
+                        my ($int_name,$unit_name) = split('\.',$name);
+                        $self->{'logger'}->debug("Checking to see if port: $name is part of circuit: $circuit_id");
+                        if(!$self->_is_circuit_on_port($circuit_id, $circuits, $int_name, $unit_name)){
                             $ri_dels .= "<instance><name>$name</name><protocols><l2vpn><site operation='delete'><name>$site_name</name></site></l2vpn></protocols></instance>";
```

Comment:
Why are we deleting the entire site instead of just removing the interface from the list?

Reply:
There's no reason that I know of. I'll try addressing that.